internal/kms/azure_vault.go

```diff
@@ -0,0 +1,230 @@
+	clientID          string
+	tenantID          string
+	clientCertificate string
+}
+
+func initAzureKeyVaultKMS(args ProviderInitArgs) (EncryptionKMS, error) {
+	kms := &azureKeyVaultKMS{
+		namespace: args.Namespace,
+	}
+
+	// required options for further configuration (getting secrets)
+	err := setConfigString(&kms.secretName, args.Config, azureSecretNameKey)
+	if errors.Is(err, errConfigOptionInvalid) {
+		return nil, err
+	} else if errors.Is(err, errConfigOptionMissing) {
+		kms.secretName = azureDefaultSecretsName
+	}
+
+	err = setConfigString(&kms.vaultURL, args.Config, azureVaultURL)
+	if err != nil {
+		return nil, err
+	}
+	err = setConfigString(&kms.clientID, args.Config, azureClientID)
+	if err != nil {
+		return nil, err
+	}
+	err = setConfigString(&kms.tenantID, args.Config, azureTenantID)
+	if err != nil {
+		return nil, err
+	}
+
+	// read the kubernetes secret with credentials
+	secrets, err := kms.getSecrets()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get secrets for %T, %w", kms, err)
+	}
+
+	err = setConfigString(&kms.clientCertificate, secrets, azureClientCertificate)
+	if err != nil {
+		return nil, err
+	}
+
+	return kms, nil
+}
+
+func (kms *azureKeyVaultKMS) Destroy() {
+	// Nothing to do.
+}
+
+func (kms *azureKeyVaultKMS) getService() (*azsecrets.Client, error) {
+	certs, key, err := azidentity.ParseCertificates([]byte(kms.clientCertificate), []byte{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse Azure client certificate: %w", err)
+	}
+	creds, err := azidentity.NewClientCertificateCredential(kms.tenantID, kms.clientID, certs, key, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create Azure credentials: %w", err)
+	}
+
+	azClient, err := azsecrets.NewClient(kms.vaultURL, creds, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create Azure client: %w", err)
+	}
+
+	return azClient, nil
+}
+
+func (kms *azureKeyVaultKMS) getSecrets() (map[string]interface{}, error) {
+	c, err := k8s.NewK8sClient()
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to kubernetes to "+
+			"get Secret %s/%s: %w", kms.namespace, kms.secretName, err)
+	}
+
+	secret, err := c.CoreV1().Secrets(kms.namespace).Get(context.TODO(),
+		kms.secretName, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get Secret %s/%s: %w", kms.namespace, kms.secretName, err)
+	}
+
+	config := make(map[string]interface{})
+	for k, v := range secret.Data {
+		switch k {
+		case azureClientCertificate:
+			config[k] = string(v)
+		default:
+			return nil, fmt.Errorf("unsupported option for KMS provider %q: %s", kmsTypeAzure, k)
+		}
+	}
+
+	return config, nil
+}
+
+// FetchDEK returns passphrase from Vault.
+func (kms *azureKeyVaultKMS) FetchDEK(key string) (string, error) {
+	svc, err := kms.getService()
+	if err != nil {
+		return "", fmt.Errorf("failed to get KMS service: %w", err)
+	}
+	// Since the second return variable loss.Version is not used, there it is ignored.
```

Comment:
Is this comment required  we are not ignoring anything here

Reply:
yea, I have removed. Yet to push latest commit. 
Thanks